srml/transaction-payment/src/lib.rs

```diff
@@ -97,7 +98,33 @@ decl_module! {
 	}
 }
 
-impl<T: Trait> Module<T> {}
+impl<T: Trait> Module<T> {
+	/// Query the data that we know about the fee of a given `call`.
+	///
+	/// As this module is not and cannot be aware of the internals of a signed extension, it only
+	/// interprets them as some encoded value and takes their length into account.
+	///
+	/// All dispatchables must be annotated with weight and will have some fee info. This function
+	/// always returns.
+	// TODO: we can actually make it understand `ChargeTransactionPayment`, but would be some hassle
```

Comment:
Do you plan to address this TODO? If not maybe let's create an issue.

Reply:
Yes I will create a potential issue but I am not sure if is even needed.. this enables the RPC to be tip-aware, while it might be waaaay too much hassle for no reason. The client itself is building the tx at the end of the day. It already knows the tip.